protocol.md

```diff
@@ -0,0 +1,233 @@
+# Mcumgr/Newtmgr protocol
```

Comment:
Since this is the repo for mcumgr, I would remove references to newtmgr except for one single place where we explain the relationship

Reply:
Also the protocol is called SMP (Simple Management Protocol)